src/bin/cargo.rs

```diff
@@ -205,8 +205,10 @@ fn execute_subcommand(config: &Config,
             }))
         }
     };
-    try!(util::process(&command).args(&args[1..]).exec());
-    Ok(())
+    match util::process(&command).args(&args[1..]).exec() {
+        Err(..) => Err(human(format!("Third party subcommand {:?} exited unsuccessfully.", command))),
```

Comment:
Perhaps this could use `chain_error` to avoid losing error information here? 

Error messages in Cargo also typically start with a lowercase letter and don't end with punctuation. I'd also recommend using a format string like:

```
third part subcommand `{}` exited unsuccessfully
```